.github/workflows/codeql.yml

```diff
@@ -0,0 +1,54 @@
+# This action runs GitHub's industry-leading static analysis engine, CodeQL, against a repository's source
+# code to find security vulnerabilities. It then automatically uploads the results to GitHub so they can be
+# displayed in the repository's security tab.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [master]
+  schedule:
+    #        ┌───────────── minute (0 - 59)
+    #        │ ┌───────────── hour (0 - 23)
+    #        │ │ ┌───────────── day of the month (1 - 31)
+    #        │ │ │ ┌───────────── month (1 - 12 or JAN-DEC)
+    #        │ │ │ │ ┌───────────── day of the week (0 - 6 or SUN-SAT)
+    #        │ │ │ │ │
+    #        │ │ │ │ │
+    #        │ │ │ │ │
+    #        * * * * *
+    - cron: '0 0 * * 4'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # Override automatic language detection by changing the below list
+        # Supported options are ['csharp', 'cpp', 'go', 'java', 'javascript', 'python']
+        language: ['java']
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Java JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
```

Comment:
Yes it could, however, we don't have a matrix since we only support java 8. Once we move this library to support Java 11 we will have to change the actions anyway. Would you like to change it from now?

Reply:
I think we might as well do it now, yeah. Will be easier to understand / extend in the future.

Reply:
Done